astropy/table/table.py

```diff
@@ -372,7 +370,9 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
         elif data is None:
             if names is None:
                 if dtype is None:
-                    return  # Empty table
+                    if meta is not None:
```

Comment:
Could not find a clean way to avoid repeating these two lines of code here and below.

Reply:
Easiest might be if we let the meta descriptor class deal with assignment of `None` (clear the whole thing). But that's definitely another PR.